dissect/target/plugins/os/unix/linux/network.py

```diff
@@ -0,0 +1,298 @@
+        """Parse virtual network configurations from systemd network configuration files."""
+
+        virtual_networks: LinuxConfigParser.VlanIdByName = {}
+        for config_file in self._config_files(self.config_paths, "*.netdev"):
+            try:
+                virtual_network_config = configutil.parse(config_file, hint="systemd")
+                net_dev_section: dict[str, str] = virtual_network_config.get("NetDev", {})
+                if net_dev_section.get("Kind") != "vlan":
+                    continue
+
+                vlan_id = virtual_network_config.get("VLAN", {}).get("Id")
+                if (name := net_dev_section.get("Name")) and vlan_id:
+                    virtual_networks[name] = int(vlan_id)
+            except Exception as e:
+                self._target.log.warning("Error parsing virtual network config file %s: %s", config_file, e)
+
+        return virtual_networks
+
+    def _parse_networks(self, virtual_networks: LinuxConfigParser.VlanIdByName) -> Iterator[UnixInterfaceRecord]:
+        """Parse network configurations from systemd network configuration files."""
+        for config_file in self._config_files(self.config_paths, "*.network"):
+            try:
+                config = configutil.parse(config_file, hint="systemd")
+
+                match_section: dict[str, str] = config.get("Match", {})
+                network_section: dict[str, str] = config.get("Network", {})
+                link_section: dict[str, str] = config.get("Link", {})
+
+                ip_interfaces: set[ip_interface] = set()
+                gateways: set[ip_address] = set()
+                dns: set[ip_address] = set()
+                mac_addresses = set[str]()
+
+                dhcp_ipv4, dhcp_ipv6 = self._parse_dhcp(network_section.get("DHCP"))
+                if link_mac := link_section.get("MACAddress"):
+                    mac_addresses.add(link_mac)
+                if match_macs := match_section.get("MACAddress"):
+                    mac_addresses.update(match_macs.split(" "))
+                if permanent_macs := match_section.get("PermanentMACAddress"):
+                    mac_addresses.update(permanent_macs.split(" "))
+
+                if dns_value := network_section.get("DNS"):
+                    if isinstance(dns_value, str):
+                        dns_value = [dns_value]
+                    dns.update({self._parse_dns_ip(dns_ip) for dns_ip in dns_value})
+
+                if address_value := network_section.get("Address"):
+                    if isinstance(address_value, str):
+                        address_value = [address_value]
+                    ip_interfaces.update({ip_interface(addr) for addr in address_value})
+
+                if gateway_value := network_section.get("Gateway"):
+                    if isinstance(gateway_value, str):
+                        gateway_value = [gateway_value]
+                    gateways.update({ip_address(gateway) for gateway in gateway_value})
```

Comment:
### With regard to `_splitter`

I re-read the documentation of `str.split()` and if you don´t specify a separator, it will separate on white space, and return an empty list if the string is empty.

### With regard to _create_list

Wrote general purpose `to_list` in `dissect/target/helpers/utils.py`